tests/components/apsystems/test_config_flow.py

```diff
@@ -50,48 +45,50 @@ async def test_form_cannot_connect_and_recover(
             },
         )
         assert result2["result"].unique_id == "MY_SERIAL_NUMBER"
-        assert result2.get("type") == FlowResultType.CREATE_ENTRY
+        assert result2.get("type") is FlowResultType.CREATE_ENTRY
         assert result2["data"].get(CONF_IP_ADDRESS) == "127.0.0.1"
 
 
-async def test_form_cannot_connect(hass: HomeAssistant, mock_setup_entry) -> None:
+async def test_form_unique_id_already_configured(
+    hass: HomeAssistant, mock_setup_entry, mock_apsystems_with_serial_id
+) -> None:
     """Test we handle cannot connect error."""
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-    with patch(
-        "homeassistant.components.apsystems.config_flow.APsystemsEZ1M",
-        return_value=AsyncMock(),
-    ) as mock_api:
-        mock_api.side_effect = TimeoutError
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN,
-            context={"source": SOURCE_USER},
-            data={
-                CONF_IP_ADDRESS: "127.0.0.2",
-            },
-        )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={
+            CONF_IP_ADDRESS: "127.0.0.2",
+        },
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["data"].get(CONF_IP_ADDRESS) == "127.0.0.2"
+    assert result["result"].unique_id == "MY_SERIAL_NUMBER"
```

Comment:
the first part of this test can be replaced by creating a `MockConfigEntry` with the right data and unique_id and adding it to hass